language/move-prover/doc/user/spec-lang.md

```diff
@@ -626,3 +626,5 @@ verification success.
 | `aborts_if_is_partial` | Allows a function to abort [under non-specified conditions](#abortsif-condition).
 | `aborts_if_is_strict`  | Disallows a function to abort even if no conditions are specified.
 | `requires_if_aborts`   | Makes a requires condition mandatory to hold even in cases where the function is specified to abort.
+| `addition_overflow_unchecked` | Makes addition of large integers (`u64` and `u128`) unchecked, avoiding the need to specify `aborts_if` for those.
+| `assume_no_abort_from_here` | Assumes that this function, if called from elsewhere, does not abort.
```

Comment:
I want to make sure I understand what `assume_no_abort_from_here` does. Is this mainly for functions like `mint` which just calls `mint_with_capability` and does nothing else? So when this pragma is on, we don't need to copy over the `aborts_if` conditions from `mint_with_capability` to `mint`?

If that's the case, having this pragma and the `addition_overflow_unchecked` pragma can save us some spec writing time. However, I'm not sure if that's a great idea in the long run as users might just ignore overflows and pay less attention to aborts?

Reply:
This is *only* to be used for functions which cause aborts based on internal errors. Check out in this PR LibraConfig::reconfigure_ for the only place where `assume_no_abort_from_here` is used. This function is not raising an error based on input parameters but on a weird internal condition.

The `addition_overflow_unchecked` is *only* to be used where say you increment a u64 by 1 or so. This happened in `Event` before I mocked it out, so currently it is not actually used. I actually first wanted to restrict this pragma so one of the parameters to add needs to be a constant, but this wasn't that easy, so its now perhaps a bit to general.

Without this, people see the need to use the pragma `aborts_if_is_partial`. This appeared absolutely deadly to me. As soon as you add this one (which I additionally did), you have no idea whether your function aborts or not. These pragmas are there to have better control on how you want to relax abort checking. We may likely want to further refine this, but it's a start.

Reply:
I've been thinking about the overflow problem.  At some future point, do you think it would be worthwhile to check some basic conditions (1) counter starts at a small value, and (2) no function causes it to increase by more than 1.  If those conditions are met, then a u64 cannot realistically overflow.

I was worrying about a case where someone makes a code change that accidentally assigns a large value to a counter or accidentally allows it to increment by a large amount, suddenly making an overflow possible.  Then, the more detailed spec would catch the change.